apps/communications/dialer/js/call_log.js

```diff
@@ -929,26 +933,40 @@ var CallLog = {
 
     // Get the list of logs to be updated.
     var logs = [];
+    var log, logInfo;
+    var i, l;
     switch (reason) {
       case 'remove':
         logs = container.querySelectorAll('li[data-contact-id="' + contactId +
                                           '"]');
+        for (i = 0, l = logs.length; i < l; i++) {
+          this._removeContact(logs[i], contactId, i === 0);
+        }
```

Comment:
I still find this really awkward and wonky passing in ```i === 0``` like this. I don't really have an alternative though. My best proposal is to create a ```_removeContacts()``` function which encapsulates this call, so that at least this call site would be cleaner. I'm open to suggestions.